### PR TITLE
Remove duplicated image processing in fiducial BQ

### PR DESCRIPTION
**Pull request by IJDykeman**
We were previously processing each image from scratch twice -- once to get the board pose and another time to get raw points.  Now we reuse the processed corner locations and IDs.

### CODE REVIEW

File: vision/fiducial_detection_and_pose.hh [hunk 1]
```diff
@@ -24,6 +24,11 @@ struct MarkerInWorld {
   int id;
 };
 
+struct board_ids_and_corners {
```
Comment (jpanikulam) [inline vision/fiducial_detection_and_pose.hh]: Conform to styleguide: CamelCase for types

File: vision/fiducial_detection_and_pose.cc [hunk 2]
```diff
@@ -34,12 +33,8 @@ std::vector<BoardPointImagePointAssociation> obj_points_img_points_from_image(co
   return result;
 }
 
-std::optional<SE3> estimate_board_center_from_camera_from_image(const cv::Mat &input_image,
+std::optional<SE3> estimate_board_center_from_camera_from_image(const board_ids_and_corners ids_corners,
```
Comment (jpanikulam) [inline vision/fiducial_detection_and_pose.cc]: Here as well

File: vision/fiducial_detection_and_pose.hh [hunk 3]
```diff
@@ -67,11 +72,13 @@ struct BoardPointImagePointAssociation {
...
 
+board_ids_and_corners get_ids_and_corners(const cv::Mat &input_image);
+
 std::vector<MarkerInWorld> get_world_from_marker_centers(const cv::Mat& camera_image, const SE3& world_from_camera);
 
-std::optional<SE3> estimate_board_center_from_camera_from_image(const cv::Mat& input_image, const Calibration& calibration);
+std::optional<SE3> estimate_board_center_from_camera_from_image(board_ids_and_corners ids_corners, const Calibration& calibration);
 
-std::vector<BoardPointImagePointAssociation> obj_points_img_points_from_image(const cv::Mat& input_image);
+std::vector<BoardPointImagePointAssociation> obj_points_img_points_from_image(board_ids_and_corners ids_corners);
```
Comment (jpanikulam) [inline vision/fiducial_detection_and_pose.hh]: `const &`

File: vision/fiducial_detection_and_pose.hh [hunk 4]
```diff
@@ -67,11 +72,13 @@ struct BoardPointImagePointAssociation {
   SERIALIZABLE_STRUCTURE(BoardPointImagePointAssociation, point_board_space, point_image_space);
 };
 
+board_ids_and_corners get_ids_and_corners(const cv::Mat &input_image);
+
 std::vector<MarkerInWorld> get_world_from_marker_centers(const cv::Mat& camera_image, const SE3& world_from_camera);
 
-std::optional<SE3> estimate_board_center_from_camera_from_image(const cv::Mat& input_image, const Calibration& calibration);
+std::optional<SE3> estimate_board_center_from_camera_from_image(board_ids_and_corners ids_corners, const Calibration& calibration);
```
Comment (jpanikulam) [inline vision/fiducial_detection_and_pose.hh]: `const &`

File: vision/fiducial_detection_and_pose.cc (hunk 2 above)
Comment (jpanikulam) [reply vision/fiducial_detection_and_pose.cc]: take as reference

File: vision/fiducial_detection_and_pose.hh (hunk 1 above)
Comment (IJDykeman) [reply vision/fiducial_detection_and_pose.hh]: *self flagellates*

**Pull request merged**